src/foreign_key/common.rs

```diff
@@ -10,6 +10,8 @@ pub struct TableForeignKey {
     pub(crate) ref_columns: Vec<DynIden>,
     pub(crate) on_delete: Option<ForeignKeyAction>,
     pub(crate) on_update: Option<ForeignKeyAction>,
+    #[cfg(any(feature = "backend-postgres", feature = "backend-sqlite"))]
+    pub(crate) deferrable: Option<Deferrable>,
```

Comment:
I think we should split `Deferrable` and `Initially` into two separate enum. i.e.

```rs
enum Deferrable { // Or represent it as a `bool`?
    Yes,
    No,
}

enum Initially {
    Deferred,
    Immediate,
}
```